src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,352 @@
+const styles = {
+  fontColor({ variant, variantColor, disabled }) {
+    switch (variant) {
+      case 'primary':
+        if (disabled) {
+          return 'light.150';
+        }
+        return 'light.100';
+      case 'secondary':
+        if (disabled) {
+          return 'shade.940';
+        }
+        if (variantColor) {
+          return `${variantColor}.800`;
+        }
+        return 'primary.800';
+      case 'tertiary':
+        if (disabled) {
+          return 'shade.940';
+        }
+        if (variantColor) {
+          return `${variantColor}.800`;
+        }
+        return 'primary.800';
+      default:
+        if (disabled) {
+          return 'light.150';
+        }
+        return 'light.100';
+    }
+  },
+  backgroundColor({ variant, buttonColor, disabled, theme }) {
+    switch (variant) {
+      case 'primary':
+        if (disabled) {
+          return getColor(theme, `${buttonColor}.500`);
+        }
+        return getColor(theme, `${buttonColor}.800`);
+      case 'secondary':
+        if (disabled) {
+          return getColor(theme, 'shade.910');
+        }
+        return getColor(theme, `${buttonColor}.920`);
+      case 'tertiary':
+        return 'transparent';
+      default:
+        if (disabled) {
+          return getColor(theme, `${buttonColor}.500`);
+        }
+        return getColor(theme, `${buttonColor}.800`);
+    }
+  },
+  border({ variant, buttonColor, disabled, theme }) {
+    switch (variant) {
+      case 'primary':
+        return '0px';
+      case 'secondary':
+        if (disabled) {
+          return `${makePxValue(0.125)} solid ${getColor(theme, 'shade.930')}`;
+        }
+        return `${makePxValue(0.125)} solid ${getColor(theme, `${buttonColor}.970`)}`;
+      case 'tertiary':
+        return '0px';
+      default:
+        return '0px';
+    }
+  },
+  underlayColor({ variant, buttonColor, theme }) {
+    switch (variant) {
+      case 'primary':
+        return getColor(theme, `${buttonColor}.600`);
+      case 'secondary':
+        return getColor(theme, `${buttonColor}.950`);
+      case 'tertiary':
+        return getColor(theme, `${buttonColor}.950`);
+      default:
+        return getColor(theme, `${buttonColor}.600`);
+    }
+  },
+  padding({ size, theme, children }) {
+    switch (size) {
+      case 'xsmall':
+        if (!children) {
+          return [theme.spacings.xxsmall];
+        }
+        return [theme.spacings.xxsmall, theme.spacings.small];
+      case 'small':
+        if (!children) {
+          return [makePxValue(0.75)];
+        }
+        return [makePxValue(0.625), theme.spacings.large];
+      case 'medium':
+        if (!children) {
+          return [theme.spacings.small];
+        }
+        return [theme.spacings.small, theme.spacings.xxlarge];
+      case 'large':
+        if (!children) {
+          return [makePxValue(1.25)];
+        }
+        return [makePxValue(1.25), theme.spacings.xxlarge];
+      default:
+        if (!children) {
```

Comment:
can we have the conditions changed? like let's prefer normal conditions over negated condtions.
instead of `if(!children)` put the `if(children)`. It's always easier to read it this way. Make this change across